cilium-dbg/cmd/encrypt_status.go

```diff
@@ -47,81 +55,132 @@ func init() {
 	command.AddOutputOption(encryptStatusCmd)
 }
 
-func getXfrmStats(mountPoint string) (int, map[string]int) {
+func getEncryptionMode() (models.EncryptionStatus, error) {
+	params := daemon.NewGetHealthzParamsWithTimeout(timeout)
+	params.SetBrief(&brief)
+	resp, err := client.Daemon.GetHealthz(params)
+	if err != nil {
+		return models.EncryptionStatus{}, err
+	}
+
+	enc := resp.Payload.Encryption
+	status := models.EncryptionStatus{
+		Mode: enc.Mode,
+	}
+	switch enc.Mode {
+	case models.EncryptionStatusModeIPsec:
+		if err := dumpIPsecStatus(&status); err != nil {
+			return models.EncryptionStatus{}, err
+		}
+	case models.EncryptionStatusModeWireguard:
+		dumpWireGuardStatus(enc, &status)
+	}
+	return status, nil
+}
+
+func dumpIPsecStatus(status *models.EncryptionStatus) error {
+	status.Ipsec = &models.IPsecStatus{}
+	xfrmStates, err := netlink.XfrmStateList(netlink.FAMILY_ALL)
+	if err != nil {
+		return fmt.Errorf("cannot get xfrm state: %s", err)
+	}
+	keys, err := ipsec.CountUniqueIPsecKeys(xfrmStates)
+	if err != nil {
+		return fmt.Errorf("error counting IPsec keys: %s\n", err)
+	}
+	status.Ipsec.KeysInUse = int64(keys)
+	decryptInts, err := getDecryptionInterfaces()
+	if err != nil {
+		return fmt.Errorf("error getting IPsec decryption interfaces: %s\n", err)
+	}
+	status.Ipsec.DecryptInterfaces = decryptInts
+	seqNum, err := maxSequenceNumber()
+	if err != nil {
+		return fmt.Errorf("error getting IPsec max sequence number: %s\n", err)
+	}
+	status.Ipsec.MaxSeqNumber = seqNum
+	errCount, errMap, err := getXfrmStats("")
+	if err != nil {
+		return fmt.Errorf("error getting xfrm stats: %s\n", err)
+	}
+	status.Ipsec.ErrorCount = errCount
+	status.Ipsec.XfrmErrors = errMap
+	return nil
+}
+
+func dumpWireGuardStatus(p *models.EncryptionStatus, status *models.EncryptionStatus) {
```

Comment:
Similar to the other comment, here I would suggest:
```suggestion
func getWireGuardStatus(p *models.EncryptionStatus) models.EncryptionStatus {
```

Reply:
Done.